### PR TITLE
drivers: eth_mcux power down

### PR DESCRIPTION
**Pull request by agansari**
Power down PHY in eth_mcux driver and set carrier off.

Signed-off-by: Andrei Gansari <andrei.gansari@nxp.com>

Depends on zephyrproject-rtos/hal_nxp#24

### CODE REVIEW

File: drivers/ethernet/eth_mcux.c [hunk 1]
```diff
@@ -295,8 +295,11 @@ static void eth_mcux_phy_event(struct eth_context *context)
 		if (context->enabled) {
 			eth_mcux_phy_enter_reset(context);
 		} else {
-			/* @todo, actually power down the PHY ? */
 			context->phy_state = eth_mcux_phy_state_initial;
+			net_eth_carrier_off(context->iface);
+			ENET_StartSMIWrite(ENET, phy_addr, PHY_BASICCONTROL_REG,
+			   kENET_MiiWriteValidFrame,
```
Comment (tbursztyka) [inline drivers/ethernet/eth_mcux.c]: aling parameters to last '(' + 1 space

File: west.yml [hunk 2]
```diff
@@ -83,7 +83,7 @@ manifest:
       revision: 00dd4fab1a00f2f6e995ef3f2e7c3814689f8885
       path: modules/fs/nffs
     - name: hal_nxp
-      revision: c6ce9b7e8ddb501ca6d54e927e1602cb5f7a7183
+      revision: 1d5f331869c32e23c3c7523881aa8f0da95f0792
```
Comment (tbursztyka) [inline west.yml]: wait that's a different change: you are bumping nxp's HAL to a newer verison. make a dedicated PR for it please.

**Review by tbursztyka — CHANGES_REQUESTED** [review]

File: west.yml (hunk 2 above)
Comment (agansari) [reply west.yml]: @tbursztyka without this version eth_mcux won't work, the version mentioned here has changes related to this patch.

Comment (tbursztyka) [reply west.yml]: So make a version bump PR first, and then the eth_mcux PR dependent over it.

Comment (agansari) [reply west.yml]: @tbursztyka wouldn't a separate commit work rather than a whole new PR, there is no major SDK update

Comment (jukkar) [reply west.yml]: Other option is to add a separate commit into this PR that updates the HAL version.

Comment (tbursztyka) [reply west.yml]: the pb with the hal version bump is that it might affect other drivers that are not related to ethernet. Thus why it's imo preferable to separate completely

Comment (jukkar) [reply west.yml]: But if we find out that the HAL version does not work everywhere we would be forced to revert both commits anyway so the end result is the same (as if we submit two separate PRs).

Comment (agansari) [reply west.yml]: @jukkar should be a good thing, one will not work without the other.

**Comment by agansari** [comment]
@jukkar @tbursztyka i've split west and driver in separate commits and aligned code.

File: west.yml (hunk 2 above)
Comment (stephanosio) [reply west.yml]: https://docs.zephyrproject.org/latest/guides/modules.html#changes-to-existing-modules

You can use `revision: pull/24/head` to run CI testing. Once it passes, the zephyrproject-rtos/hal_nxp#24 can be merged and the revision here can be updated to the commit hash in the master (i.e. the commit that has been rebased onto master).

**Comment by jukkar** [comment]
Is this PR abandoned?
